buildSrc/src/main/kotlin/otel.publish-conventions.gradle.kts

```diff
@@ -0,0 +1,103 @@
+plugins {
+    `maven-publish`
+    signing
+
+    id("otel.japicmp-conventions")
+}
+
+publishing {
+    publications {
+        register<MavenPublication>("mavenPublication") {
+            val release = findProperty("otel.release")
+            if (release != null) {
+                val versionParts = version.split('-').toMutableList()
+                versionParts[0] += "-${release}"
+                version = versionParts.joinToString("-")
+            }
+            groupId = "io.opentelemetry"
+            afterEvaluate {
+                // not available until evaluated.
+                artifactId = base.archivesName.get()
+                pom.description.set(project.description)
+            }
+
+            plugins.withId("java-platform") {
+                from(components["javaPlatform"])
+            }
+            plugins.withId("java-library") {
+                from(components["java"])
+            }
+
+            versionMapping {
+                allVariants {
+                    fromResolutionResult()
+                }
+            }
+
+            pom {
+                name.set("OpenTelemetry Java")
+                url.set("https://github.com/open-telemetry/opentelemetry-java")
+
+                licenses {
+                    license {
+                        name.set("The Apache License, Version 2.0")
+                        url.set("http://www.apache.org/licenses/LICENSE-2.0.txt")
+                    }
+                }
+
+                developers {
+                    developer {
+                        id.set("opentelemetry")
+                        name.set("OpenTelemetry")
+                        url.set("https://github.com/open-telemetry/community")
+                    }
+                }
+
+                scm {
+                    connection.set("scm:git:git@github.com:open-telemetry/opentelemetry-java.git")
+                    developerConnection.set("scm:git:git@github.com:open-telemetry/opentelemetry-java.git")
+                    url.set("git@github.com:open-telemetry/opentelemetry-java.git")
+                }
+            }
+        }
+    }
+}
+
+afterEvaluate {
+    val publishToSonatype by tasks.getting
+    val release by rootProject.tasks.existing
+    release.configure {
+        finalizedBy(publishToSonatype)
+    }
+}
+
+if (System.getenv("CI") != null) {
+    signing {
+        useInMemoryPgpKeys(System.getenv("GPG_PRIVATE_KEY"), System.getenv("GPG_PASSWORD"))
+        sign(publishing.publications["mavenPublication"])
+    }
+}
+
+tasks {
+    register("updateVersionInDocs") {
```

Comment:
We previously added this to allprojects but it seems to only be needed for published ones